feature-016 : 맛집 평가 생성(API)

Chestnut90 opened the pull request:
## PR 체크리스트
아래 항목을 확인해 주세요:

- [x] 커밋 메시지가 우리의 가이드라인을 따르고 있는지 확인하세요
- [x] 변경 사항에 대한 테스트가 추가되었는지 확인하세요 (버그 수정 / 기능 추가)
- [ ] 문서가 추가되거나 업데이트되었는지 확인하세요 (버그 수정 / 기능 추가)

## PR 유형
이 PR은 어떤 종류의 변경을 가져오나요?

<!-- 이 PR에 해당하는 것을 "x"로 표시하세요. -->

- [ ] 버그 수정
- [x] 새로운 기능 추가
- [ ] 코드 스타일 업데이트 (서식, 로컬 변수)
- [ ] 리팩터링 (기능 변경 없음, API 변경 없음)
- [ ] 빌드 관련 변경
- [ ] CI 관련 변경
- [ ] 문서 내용 변경
- [ ] 애플리케이션 / 인프라 변경
- [ ] 기타... 설명:

## 현재 동작은 무엇인가요?
<!-- 수정 중인 현재 동작을 설명하거나 관련된 이슈에 링크하세요. -->

이슈 번호: #16 

## 새로운 동작은 무엇인가요?

## 이 PR은 호환성 변경을 도입하나요?

- [ ] 예
- [x] 아니요

<!-- 이 PR에 호환성 변경이 포함되어 있다면, 기존 응용 프로그램에 대한 영향과 마이그레이션 경로를 아래에 설명해 주세요. -->

## 기타 정보
<img width="728" alt="image" src="https://github.com/Django-Wanted-Internship-3-Team/repo2_restaurant_recommend/assets/22581530/520de6d2-a362-4112-898f-76190a3e4f86">

Chestnut90 committed:
:recycle:Refact : Review model 리팩터 	#16

- rating의 default=0 및 0~5 constraints 추가
- content의 length 255 및 default="" 추가

Related to #16

Chestnut90 committed:
:white_check_mark:Test : Review api reverse 테스트 케이스 #16

- 현재 review api가 존재하지 않음을 테스트

Related to #16

Chestnut90 committed:
:sparkles:Feat : Review API skeleton 생성 및 url 연결 #16

- Review post api 에 대한 APIView 생성
- api exists 테스트 통과

Related to #16

Chestnut90 committed:
:white_check_mark:Test : Review api에 대한 auth 상태의 테스트 케이스 추가 #16

- Review api는 사용자 인증 후 동작하기에 인증된 상태와 인증되지 않은 상태에서 테스트

Related to #16

Chestnut90 committed:
:recycle:Refact : Review api 리팩터링 #16

- api 테스트에 대한 구현
- APIView post 구현을 통해 review 등록

Related to #16

Chestnut90 committed:
:recycle:Refact : Restaurant and Model Migration #16

Related to #16

Chestnut90 committed:
:white_check_mark:Test : Review api 옳바르지 않은 rating 값 테스트 케이스 #16

- rating은 0~5사이의 값이어야 한다.
Related to #16

Chestnut90 committed:
:recycle:Refact : Review 의 rating 검증 추가 #16

- review의 rating값이 0~5인 값만 취급
Related to #16

Chestnut90 committed:
Merge branch 'develop' into feature/issue-016

Chestnut90 committed:
:recycle:Refact : mypy type check

Chestnut90 committed:
:recycle:Refact : flake8 요구사항 맞춤

saJaeHyukc commented on restaurants_recommendation/reviews/models.py:
```diff
@@ -1,16 +1,20 @@
 class Review(models.Model):
-    rating = models.FloatField(null=True)
-    content = models.CharField(max_length=128)
+    rating = models.FloatField(default=0)
```
constarints 부분이 값지정하기 위하여 설정한 것으로 확인됩니다!
field범위 내에서 아래와 같은 방법도 있을 것 같습니다 :) 

```suggestion
from django.core.validators import MaxValueValidator, MinValueValidator

rating = models.FloatField(
    validators=[MinValueValidator(0.0), MaxValueValidator(5.0)],
)
```

Chestnut90 replied:
좋은 정보 감사합니다. 
테스트를 해보니 validator를 사용하면 필드 자체에 값을 검사하기 때문에 serializer에서 rating에 대한 validation을 따로 수행하지 않아도 되네요!
다만 validator를 구성하면 python level에서만 동작하는데, 실무에서는 Database에 어느정도까지 제약을 추가하는지가 궁금하네요!

saJaeHyukc replied:
평점이 중요한 것에 따라 다를 것 같습니다 :)

simseulnyang reviewed (approved):
코드 확인했습니다! 수고하셨습니다🙂

lfoyh6591 commented:
LGTM :)

Chestnut90 committed:
:recycle:Refact : Review의 rating에 validator 사용 #16

- rating의 값 구간을 설정하기 위해서 Min, Max validator 사용
- ReviewSerializer의 validate_rating 제거 -> model의 fields에서 수행하기 때문

Related to #16
